test/OpenSpaceToolkit/Astrodynamics/Flight/Profile.test.cpp

```diff
@@ -883,6 +889,68 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile, OrientationProfileTarget)
     }
 }
 
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile, YawCompensation)
+{
+    const Environment environment = Environment::Default();
+
+    const Length semiMajorAxis = Length::Kilometers(7000.0);
+    const Real eccentricity = 0.0;
+    const Angle inclination = Angle::Degrees(98.0);
+    const Angle raan = Angle::Degrees(0.0);
+    const Angle aop = Angle::Degrees(0.0);
+    const Angle trueAnomaly = Angle::Degrees(0.0);
+
+    const COE coe = {semiMajorAxis, eccentricity, inclination, raan, aop, trueAnomaly};
+
+    const Instant instant = Instant::DateTime(DateTime(2018, 1, 1, 0, 0, 0), Scale::UTC);
+    const Derived gravitationalParameter = EarthGravitationalModel::EGM2008.gravitationalParameter_;
+    const Length equatorialRadius = EarthGravitationalModel::EGM2008.equatorialRadius_;
+    const Real J2 = EarthGravitationalModel::EGM2008.J2_;
+    const Real J4 = EarthGravitationalModel::EGM2008.J4_;
+
+    const Kepler keplerianModel = {
+        coe, instant, gravitationalParameter, equatorialRadius, J2, J4, Kepler::PerturbationType::None
+    };
+
+    const Orbit orbit = {keplerianModel, environment.accessCelestialObjectWithName("Earth")};
+
+    const State state = orbit.getStateAt(instant);
+
+    const Array<Instant> instants =
+        Interval::Closed(instant, instant + Duration::Seconds(2.0)).generateGrid(Duration::Seconds(1.0));
+
+    const Trajectory trajectory = Trajectory::GroundStripGeodeticNadir(orbit, instants, Earth::WGS84());
+
+    const Profile compensatedProfile = Profile::CustomPointing(
+        orbit,
+        std::make_shared<const Profile::TrajectoryTarget>(
+            Profile::TrajectoryTarget::TargetPosition(trajectory, Profile::Axis::Z)
+        ),
+        std::make_shared<const Profile::TrajectoryTarget>(
+            Profile::TrajectoryTarget::TargetVelocity(trajectory, Profile::Axis::X)
+        )
+    );
+
+    const Profile uncompensatedProfile = Profile::CustomPointing(
+        orbit,
+        std::make_shared<const Profile::TrajectoryTarget>(
+            Profile::TrajectoryTarget::TargetPosition(trajectory, Profile::Axis::Z)
+        ),
+        std::make_shared<const Profile::Target>(Profile::TargetType::VelocityECI, Profile::Axis::X)
+    );
+
+    const State compensatedState = compensatedProfile.getStateAt(instant);
+    const State uncompensatedState = uncompensatedProfile.getStateAt(instant);
+
+    // Values taken from Orekit for unit test
+
+    EXPECT_NEAR(
+        compensatedState.getAttitude().angularDifferenceWith(uncompensatedState.getAttitude()).inDegrees(),
+        3.803545407107513,
+        1e-4
+    );
```

Comment:
can you assert a positive angular differnce pointing westwards or something like that? Like to physically assert that the clocking axis (X here) has a westwards component?

Reply:
I can just assert the quaternion lines up exactly as Orekit.